src/particles/beam/BeamParticleContainer.cpp

```diff
@@ -65,6 +65,23 @@ BeamParticleContainer::ReadParameters ()
     AMREX_ALWAYS_ASSERT_WITH_MESSAGE( m_n_subcycles >= 1, "n_subcycles must be >= 1");
     queryWithParserAlt(pp, "do_reset_id_init", m_do_reset_id_init, pp_alt);
     queryWithParser(pp, "do_salame", m_do_salame);
+    amrex::Array<std::string, 3> field_str = {"0", "0", "0"};
+    m_use_external_fields = queryWithParserAlt(pp, "external_E(x,y,z,t)", field_str, pp_alt);
+    m_external_fields[0] = makeFunctionWithParser<4>(field_str[0], m_external_fields_parser[0],
+        {"x", "y", "z", "t"});
+    m_external_fields[1] = makeFunctionWithParser<4>(field_str[1], m_external_fields_parser[1],
+        {"x", "y", "z", "t"});
+    m_external_fields[2] = makeFunctionWithParser<4>(field_str[2], m_external_fields_parser[2],
+        {"x", "y", "z", "t"});
+    field_str = {"0", "0", "0"};
+    m_use_external_fields = queryWithParserAlt(pp, "external_B(x,y,z,t)", field_str, pp_alt)
+        || m_use_external_fields;
```

Comment:
Just a suggestion, I have no preference.
```suggestion
    m_use_external_fields += queryWithParserAlt(pp, "external_B(x,y,z,t)", field_str, pp_alt);
```

Reply:
Using `+=` for bool will generate a warning on some compilers, for example MSVC.

Reply:
Thanks, I was not sure about that.